### PR TITLE
chore: use workspace dependencies

### LINKED ISSUE #3123: Convert commonly used dependencies into workspace style deps
### Describe the feature

we already have this for revm deps, we should do this for all commonly used dependencies

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by 0xJepsen**
These were the top 21 most common dependancies, I moved them to workspace dependancies. The number is the frequency of use across all crates. I didn't move any used less than 6 times. 
Closes #3123
```
reth-primitives: 31
thiserror: 20
reth-interfaces: 16
reth-provider: 15
reth-rlp: 13
tokio-stream: 13
tokio: 26
serde: 11
reth-rpc-types: 10
async-trait: 10
reth-metrics: 9
futures: 8
serde_json: 8
secp256k1: 8
reth-payload-builder: 7
pin-project: 7
reth-transaction-pool: 6
futures-util: 6
reth-tasks: 6
rand: 6
reth-network-api: 6
tokio-util: 6
```

### CODE REVIEW

**Review by onbjerg — APPROVED** [review]
lgtm, waiting for another approval

**Review by mattsse — APPROVED** [review]
nice

lgtm.

**Pull request merged**
